File: app/java/src/main/java/com/example/production/domain/model/purchase/Acceptance.java
package com.example.production.domain.model.purchase;

import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 検収データ
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Acceptance {
    private Integer id;
    private String acceptanceNumber;
    private String inspectionNumber;
    private String purchaseOrderNumber;
    private Integer lineNumber;
    private LocalDate acceptanceDate;
    private String acceptorCode;
    private String supplierCode;
    private String itemCode;
    private Boolean miscellaneousItemFlag;
    private BigDecimal acceptedQuantity;
    private BigDecimal unitPrice;
    private BigDecimal amount;
    private BigDecimal taxAmount;
    private String remarks;
    private LocalDateTime createdAt;
    private String createdBy;
    private LocalDateTime updatedAt;
    private String updatedBy;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Purchase/Acceptance.cs
namespace ProductionManagement.Domain.Models.Purchase;

/// <summary>
/// 検収データ
/// </summary>
public class Acceptance
{
    public int Id { get; set; }
    public required string AcceptanceNumber { get; set; }
    public required string InspectionNumber { get; set; }
    public required DateOnly AcceptanceDate { get; set; }
    public required decimal AcceptanceQuantity { get; set; }
    public required decimal AcceptanceAmount { get; set; }
    public decimal TaxAmount { get; set; }
    public string? StorageLocationCode { get; set; }
    public string? Remarks { get; set; }
    public DateTime? CreatedAt { get; set; }
    public string? CreatedBy { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public string? UpdatedBy { get; set; }
}
